### PR TITLE
ovl_Elf_Msg decompiled

### PR DESCRIPTION
**Pull request by sonicdcer**

### CODE REVIEW

File: src/overlays/actors/ovl_Elf_Msg/z_elf_msg.c [hunk 1]
```diff
@@ -29,30 +30,121 @@ const ActorInit Elf_Msg_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg/func_8092E1D0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg/func_8092E1FC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg/func_8092E284.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg/ElfMsg_Update.s")
+void ElfMsg_SetupAction(ElfMsg* this, ElfMsgActionFunc actionFunc) {
+    (void)"共倒れ";
+    (void)"共倒れ";
```
Comment (hensldm) [inline src/overlays/actors/ovl_Elf_Msg/z_elf_msg.c:40]: Same comment as in #865

File: src/overlays/actors/ovl_Elf_Msg/z_elf_msg.c [hunk 2]
```diff
@@ -29,30 +30,121 @@ const ActorInit Elf_Msg_InitVars = {
...
+        (Flags_GetSwitch(play, this->actor.home.rot.y - 1))) {
+        if (ENELF_GET_7F00(&this->actor) != 0x7F) {
+            Flags_SetSwitch(play, ENELF_GET_7F00(&this->actor));
+        }
+        Actor_MarkForDeath(&this->actor);
+        return true;
+    }
+    if (this->actor.home.rot.y == 0x81) {
+        play = play;
+        this = this;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Elf_Msg/z_elf_msg.c:55]:
```suggestion
```
Needed?

File: src/overlays/actors/ovl_Elf_Msg/z_elf_msg.c [hunk 3]
```diff
@@ -29,30 +30,121 @@ const ActorInit Elf_Msg_InitVars = {
...
+        if (Flags_GetClear(play, this->actor.room)) {
+            if (ENELF_GET_7F00(&this->actor) != 0x7F) {
+                Flags_SetSwitch(play, ENELF_GET_7F00(&this->actor));
+            }
+            Actor_MarkForDeath(&this->actor);
+            return true;
+        }
+    }
+    if ((this->actor.params & 0x7F00) >> 8 == 0x7F) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Elf_Msg/z_elf_msg.c:64]:
```suggestion
    if (ENELF_GET_7F00(&this->actor) == 0x7F) {
```

File: src/overlays/actors/ovl_Elf_Msg/z_elf_msg.c [hunk 4]
```diff
@@ -29,30 +30,121 @@ const ActorInit Elf_Msg_InitVars = {
...
+        Actor_MarkForDeath(&this->actor);
+        return true;
+    }
+    return false;
+}
+
+void ElfMsg_Init(Actor* thisx, PlayState* play) {
+    ElfMsg* this = THIS;
+
+    if (func_8092DF9C(this, play) == 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Elf_Msg/z_elf_msg.c:77]:
```suggestion
    if (!func_8092DF9C(this, play)) {
```

File: src/overlays/actors/ovl_Elf_Msg/z_elf_msg.c [hunk 5]
```diff
@@ -29,30 +30,119 @@ const ActorInit Elf_Msg_InitVars = {
...
+    }
+}
+
+void ElfMsg_Update(Actor* thisx, PlayState* play) {
+    ElfMsg* this = THIS;
+
+    if (func_8092DF9C(this, play) == 0) {
+        if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+            if (ENELF_GET_7F00(thisx) != 0x7F) {
+                Flags_SetSwitch(play, ENELF_GET_7F00(thisx));
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Elf_Msg/z_elf_msg.c:140]: Can be GET_SWITCH then.

**Review by EllipticEllipsis — APPROVED** [review]
Wow these look similar.

**Review by hensldm — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**